src/uct/ib/mlx5/dc/dc_mlx5_ep.c

```diff
@@ -23,6 +23,16 @@ uct_dc_mlx5_add_flush_remote(uct_dc_mlx5_ep_t *ep)
     ep->flags |= UCT_DC_MLX5_EP_FLAG_FLUSH_REMOTE;
 }
 
+static UCS_F_ALWAYS_INLINE uint16_t
+uct_dc_mlx5_atomic_offset(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep)
+{
+    if (uct_ib_iface_md(&iface->super.super.super)->config.enable_indirect_atomic) {
```

Comment:
seems wrong - we can't decide to use offset 0 without also using direct (and not atomic) part of the rkey

Reply:
if enable_indirect_atomic is off the atomic part of the rkey won't have an offset but may still be present to enforce strict order

Reply:
can we detect it from information provided from the target side, rather than local configuration?

Reply:
check if iface/ep address can have extra flag to indicate if atomic offset is 0, or it's as defined by atomic_mr_id